src/app/views/App.tsx

```diff
@@ -41,17 +40,25 @@ interface IAppProps {
 
 interface IAppState {
   selectedVerb: string;
+  showSidebar: boolean;
+  showToggle: boolean;
 }
 
 class App extends Component<IAppProps, IAppState> {
   constructor(props: IAppProps) {
     super(props);
     this.state = {
-      selectedVerb: 'GET'
+      selectedVerb: 'GET',
+      showSidebar: true,
+      showToggle: false,
     };
   }
 
   public componentDidMount = () => {
+    const mediaQueryList = window.matchMedia('(max-width: 767px)');
+    this.displayToggleButton(mediaQueryList);
+    mediaQueryList.addListener(this.displayToggleButton);
```

Comment:
Make sure to remove the listener in `componentWillUnMount`. Not removing listeners is a common cause for memory leaks.